common/src/main/java/com/wynntils/models/items/items/game/GearItem.java

```diff
@@ -33,13 +37,23 @@ public class GearItem extends GameItem
                 PowderedItemProperty,
                 RerollableItemProperty,
                 ShinyItemProperty,
-                IdentifiableItemProperty<GearInfo, GearInstance> {
+                IdentifiableItemProperty<GearInfo, GearInstance>,
+                SetItemProperty,
+                RequirementItemProperty {
     private final GearInfo gearInfo;
     private final GearInstance gearInstance;
-
-    public GearItem(GearInfo gearInfo, GearInstance gearInstance) {
+    private final Optional<SetInfo> setInfo;
+    private final Optional<SetInstance> setInstance;
```

Comment:
And now, let's address `SetInstance`. It took me a while to get a grip on the issue, but I think I basically got it now.

Let's remove `trueRoll` from `SetInstance`. That means that you can populate `SetInstance` when parsing everything `GearInstance` related, purely from the tooltip (even if you get wrong data when joining a world, items are going to get resent shortly, with correct tooltips).  You can remove every hacky, reference-based changes (if there is any left) in `SetModel` too.  While you are at it, move `Optional<SetInstance>` into `GearInstance`.

This means that most of the changes are gone, but we still have set instance data. For actually getting the true roll data, get it through `SetModel`. This should also mean that whenever Wynn fixes this bug, we only have to remove parts of the model, without touching anything else.